### PR TITLE
Sorr task592 extend unit testing of str to timestamp

### PR DESCRIPTION
**Pull request by shab95**
Task #592 
- changed docstring in the `str_to_timestamp()` function
- added UTC timezone unit test

### CODE REVIEW

File: helpers/test/test_hdatetime.py [hunk 1]
```diff
@@ -673,3 +673,14 @@ def test4(self) -> None:
         actual = str(err.exception)
         expected = "Unknown string format: qwe28abc07-201234 present at position 0"
         self.assert_equal(actual, expected)
+
+    def test5(self) -> None:
+        """
+        Test valid datetime with UTC timezone.
```
Comment (rheenina) [inline helpers/test/test_hdatetime.py:679]: @DanilYachmenev wdyt if we just change timezone in the `test2()`? `test5()` looks like the `test1()` except one line.

File: helpers/hdatetime.py [hunk 2]
```diff
@@ -530,6 +530,11 @@ def str_to_timestamp(
     """
     Convert timestamp as string to `pd.Timestamp`.
 
+    The function localizes input time to the specified timezone,
```
Comment (rheenina) [inline helpers/hdatetime.py:533]: ->
Localize input time to the specified timezone.

E.g., \`timestamp_as_str = "20230523_150513"\`:
\- \`tz = "UTC"\` -> "2023-05-23 15:05:13+0000"
\- \`tz = "US/Eastern"\` -> "2023-05-23 15:05:13-0400"

**Review by rheenina — COMMENTED** [review]
GJ! only nits

File: helpers/test/test_hdatetime.py (hunk 1 above)
Comment (shab95) [reply helpers/test/test_hdatetime.py]: I implemented this for now, can revert if needed later.
@DanilYachmenev

**Comment by shab95** [comment]
@rheenina Just finished up some of the things you mentioned in #590. I didn't resolve them just so you can still see them. Not sure what the right protocol is here, sorry if I pinged too much.

File: helpers/test/test_hdatetime.py [hunk 3]
```diff
@@ -626,84 +626,88 @@ def test3(self) -> None:
...
             )
         actual = str(err.exception)
-        expected = "time data '28-07-2023 15:05:13' does not match format '%Y%m%d_%H%M%S' (match)"
-        self.assert_equal(actual, expected)
+        self.check_string(actual)
 
     def test4(self) -> None:
         """
-        Test invalid datetime without format.
+        T- `datetime_str` has an invalid format
```
Comment (gpsaggese) [inline helpers/test/test_hdatetime.py:677]: Typo

File: helpers/test/test_hdatetime.py (hunk 1 above)
Comment (shab95) [reply helpers/test/test_hdatetime.py]: @DanilYachmenev

Comment (DanilYachmenev) [reply helpers/test/test_hdatetime.py]: > @DanilYachmenev wdyt if we just change timezone in the `test2()`? `test5()` looks like the `test1()` except one line.

agree!

thx for ping @shab95, missed this one
pls do the change in `test2()` instead

**Comment by shab95** [comment]
@rheenina Just wanted to put this review on your radar.

File: helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt [hunk 4]
```diff
@@ -1 +1 @@
-abc
```
Comment (rheenina) [inline helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt:1]: could you pls revert this files to master using `git checkout master {file_name}`
helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt
helpers/test/outcomes/TestCheckString1.test_check_string1/output/test.txt
helpers/test/outcomes/TestCheckString1.test_check_string_not_equal1/output/test.txt
helpers/test/outcomes/TestCheckString1.test_check_string_not_equal2/output/test.txt
helpers/test/outcomes/TestCheckString1.test_check_string_not_equal3/output/test.txt

That's the usual thing that happens when tests are running on a machine with a different OS.

Comment (shab95) [reply helpers/test/outcomes/TestCheckDataFrame1.test_check_df_not_equal3/output/test_df.txt]: Sorry about this, I have a bad habit of staging everything on accident. Just reverted this.

File: helpers/test/test_hdatetime.py [hunk 5]
```diff
@@ -626,84 +626,88 @@ def test3(self) -> None:
...
 # #############################################################################
 # Test_dassert_str_is_date
 # #############################################################################
 
 
 class Test_dassert_str_is_date(hunitest.TestCase):
+    """
+    Test that the function checks a string representation of date correctly.
+    """
     def test1(self) -> None:
```
Comment (rheenina) [inline helpers/test/test_hdatetime.py:698]: 1 blank line after a test class doc string

File: helpers/test/test_hdatetime.py [hunk 6]
```diff
@@ -626,84 +626,88 @@ def test3(self) -> None:
 
 
 class Test_str_to_timestamp1(hunitest.TestCase):
+    """
+    Test if string representation of datetime is converted correctly
+    """
     def test1(self) -> None:
```
Comment (rheenina) [inline helpers/test/test_hdatetime.py:632]: 1 blank line after a test class doc string

**Review by rheenina — APPROVED** [review]
LGTM, merging FYI @DanilYachmenev

**Pull request merged**